scylla-cdc-base/src/main/java/com/scylladb/cdc/model/worker/Worker.java

```diff
@@ -89,9 +98,29 @@ private static TaskState getInitialStateForStreams(Map<TaskId, SortedSet<StreamI
      * This includes fetching saved state of each task or creating a new initial
      * state for tasks that haven't run successfully before.
      */
-    private TaskActionsQueue queueFirstActionForEachTask(Map<TaskId, SortedSet<StreamId>> groupedStreams) throws ExecutionException, InterruptedException {
-        return new TaskActionsQueue(createTasksWithState(groupedStreams)
-                .map(task -> TaskAction.createFirstAction(workerConfiguration, task)).collect(Collectors.toSet()));
+    private Collection<TaskAction> queueFirstActionForEachTask(Map<TaskId, SortedSet<StreamId>> groupedStreams)
```

Comment:
Probably the name of the method needs to be changed.

Reply:
Why? It still queues the initial actions. Just because the data type is not named "queue" does not make the result less of one.

Reply:
Because now it's more like collecting tasks not yet queueing them yet. They are queued for execution in the caller of this method.

Reply:
The collection is effectively a queue. Just because the queue is then applied to another queue...